compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -290,7 +290,13 @@ impl<'f> PerFunctionContext<'f> {
                     references.aliases.insert(Expression::Other(result), AliasSet::known(result));
                     references.set_known_value(result, address);
 
-                    self.last_loads.insert(address, (instruction, block_id));
+                    self.load_results.insert(result, PerFuncLoadResultContext::new(instruction));
+
+                    let load_counter = self
+                        .last_loads
+                        .get(&address)
+                        .map_or(1, |(_, _, load_counter)| *load_counter + 1);
+                    self.last_loads.insert(address, (instruction, block_id, load_counter));
```

Comment:
```suggestion
                    let last_load = self
                        .last_loads
                        .entry(address)
                        .or_insert((instruction, block_id, 0));
                    last_load.load_counter += 1;
```
Could probably fix the formatting with an intermediate variable if this version is too long.

Reply:
Nice yeah it did improve the formatting

Reply:
Ah so I switched this to `PerFuncLastLoadContext::new` and we have the multi-line formatting again. I kept this logic either way though.

Reply:
I actually realized that this breaks the last loads functionality as we always want to reinsert the last load that we have, so I switched back to the map_or.

Reply:
Why do we want to reinsert? I don't see the difference between the map_or version and the or_insert version above

Reply:
We use the block_id of the last load to determine whether the last load is in the return block. I could do `last_load.block_id = block_id` with the counter increment, but I found the map_or cleaner for this purpose.

Reply:
Oh, I see we always mutate the instruction and block id. Was missing that thinking we only did it when inserting a new item.